mne/decoding/time_gen.py

```diff
@@ -347,41 +410,34 @@ def score(self, epochs=None, y=None):
                                  'for scoring.')
         elif isinstance(y, list):
             y = np.array(y)
+
+        # Clean attributes
+        for att in ['scores_', 'y_true_']:
+            if hasattr(self, att):
+                delattr(self, att)
+
         self.y_true_ = y  # to be compared with y_pred for scoring
 
         # Preprocessing for parallelization
         n_jobs = min(len(self.y_pred_[0][0]), check_n_jobs(self.n_jobs))
-        parallel, p_time_gen, n_jobs = parallel_func(_score_slices, n_jobs)
+        parallel, p_func, n_jobs = parallel_func(_score_slices, n_jobs)
         n_estimators = len(self.train_times_['slices'])
         n_chunks = min(n_estimators, n_jobs)
-        splits = np.array_split(range(len(self.train_times_['slices'])),
+        chunks = np.array_split(range(len(self.train_times_['slices'])),
                                 n_chunks)
-        scores = parallel(
-            p_time_gen(self.y_true_,
-                       [self.y_pred_[train] for train in split],
-                       self.scorer_)
-            for split in splits)
-
+        scores = parallel(p_func(
+            self.y_true_, [self.y_pred_[train] for train in chunk],
+            self.scorer_) for chunk in chunks)
```

Comment:
The parallelization is training times, but to avoid overheads, we divide them in large chunks.